### PR TITLE
Add a test that implicit submission is not allowed when the first submit button is disabled.

### PR DESCRIPTION
**Pull request by rniwa**

### CODE REVIEW

**Comment by rniwa** [comment]
Blink and Gecko already passes the test. WebKit will once https://github.com/WebKit/WebKit/pull/3093 lands.

File: html/semantics/forms/form-submission-0/implicit-submission.optional.html [hunk 1]
```diff
@@ -35,5 +35,15 @@
   assert_equals(event.submitter, null);
   assert_true(event instanceof SubmitEvent);
 }, 'Submit event with no submit button');
+
+promise_test(async () => {
+  let form = populateForm('<input name=text value=abc><input name=submitButton type=submit disabled>');
+  let event;
+  form.text.focus();
+  form.addEventListener('submit', e => { event = e; });
```
Comment (nt1m) [inline html/semantics/forms/form-submission-0/implicit-submission.optional.html:43]: I suggest using `assert_unreached()`: https://web-platform-tests.org/writing-tests/testharness-api.html?highlight=unreach#assert_unreached

This really provides a guarantee that the event listener is never reached.

Comment (rniwa) [reply html/semantics/forms/form-submission-0/implicit-submission.optional.html:43]: Hm... with it, the test passes and spits out a separate harness error message. I think that's worse than what I have right now, which shows clear "FAIL" for the test case.
```
CONSOLE MESSAGE: Error: assert_unreached: Reached unreachable code





Harness Error (FAIL), message = Error: assert_unreached: Reached unreachable code

PASS Submit event with a submit button
PASS Submit event with no submit button
PASS Submit event with disabled submit button
```

**Review by nt1m — APPROVED** [review]

File: html/semantics/forms/form-submission-0/implicit-submission.optional.html (hunk 1 above)
Comment (domenic) [reply html/semantics/forms/form-submission-0/implicit-submission.optional.html:43]: I think best would be using `t.unreached_func()` (you'd need to add a `t` parameter).

Comment (rniwa) [reply html/semantics/forms/form-submission-0/implicit-submission.optional.html:43]: Cool, `unreached_func` works pretty well here.

**Pull request merged**